### ISSUE TITLE
Connection error after executing a long step

### ISSUE THREAD

**Issue by hakito** [getgauge/gauge-csharp#57]
When a step takes longer than 350 seconds, gauge crashes with a connection error:

```
C:\work\gauge-bug\csharp>gauge specs
Build Building Project: C:\work\gauge-bug\csharp\csharp.sln
... snip ...
Build Success
# Timeout
  ## Timeout    Starting sleep of 350 seconds
Slept for a second, elapsed 0
... snip ...
Slept for a second, elapsed 349
Sleep finished
 F
        Error Message: Connection closed [127.0.0.1:50049] cause: EOF
        Stacktrace:
... snip ...
Build System.Runtime.Remoting.RemotingException: Cannot load type 'Gauge.CSharp.Runner.ISandbox, Gauge.CSharp.Runner, Version=0.7.1.0, Culture=neutral, PublicKeyToken=null'.
... snip ...
Specifications: 1 executed      0 passed        1 failed        0 skipped
Scenarios:      1 executed      0 passed        1 failed        0 skipped
```

I tried it with different time values:

* 305 seconds works
* 310 seconds crashes

I also tested the java runner with 600 seconds, and didn't get a connection error. So I think the issue is only related to the csharp runner.

[stdout.txt](https://github.com/getgauge/gauge-csharp/files/165037/stdout.txt)
[csharp.zip](https://github.com/getgauge/gauge-csharp/files/165040/csharp.zip)

**Comment by sriv** [comment]
Released via [0.7.2](https://github.com/getgauge/gauge-csharp/releases/tag/v0.7.2)

**Comment by sguptatw** [comment]
Where would one write a functional test for this?
